com.ibm.streamsx.topology/opt/python/include/splpy.h

```diff
@@ -184,6 +184,37 @@ namespace streamsx {
       return ret;
     }
 
+    static std::auto_ptr<SPL::int32> pyTupleHash(PyObject * function, PyObject * pickleObjectFunction, SPL::blob & pyblob) {
+
+      std::auto_ptr<SPL::int32> ret;
+      PyGILLock lock;
+
+      // convert spl blob to bytes
+      PyObject * pyBytes  = pyBlobToBytes(pyblob);
+      // invoke python nested function that calls the application function
+      PyObject * pyReturnVar = pyTupleFunc(function, pyBytes);
+      if (pyReturnVar == Py_None){
```

Comment:
Could we have some comments explaining what is going on?

I would have thought getting a hash was a simple situation, every object has a hash but there is a three way return from some python function, 0 None or something else.